lightning_client.go

```diff
@@ -3042,3 +3049,36 @@ func (s *lightningClient) SubscribeInvoices(ctx context.Context,
 
 	return invoiceUpdates, streamErr, nil
 }
+
+type CheckMacPermRequest struct {
+	Macaroon string
+
+	Permissions []bakery.Op
+}
+
+func (s *lightningClient) CheckMacaroonPermissions(ctx context.Context,
```

Comment:
I think for this we don't need an extra type (`CheckMacPermRequest`). Could just have the method signature like this:
```
CheckMacaroonPermissions(ctx context.Context, macaroon []byte, permissions []MacaroonPermission) (
		bool, error)
```

The type `MacaroonPermission` is available on the `lnd-13-0` branch (see my main comment).